benchmarks/allocator_no-op_benchmarks.cpp

```diff
@@ -0,0 +1,68 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "benchmark/benchmark.h"
+#include "umpire/config.hpp"
+#include "umpire/Allocator.hpp"
+#include "umpire/ResourceManager.hpp"
+
+static const int RangeLow{4};
+static const int RangeHi{1024};
+
+class NoOpAllocatorBenchmark : public benchmark::Fixture
+{
+public:
+  using ::benchmark::Fixture::SetUp;
+  using ::benchmark::Fixture::TearDown;
+
+  NoOpAllocatorBenchmark() {}
+
+  void SetUp(benchmark::State&) {
+    auto& rm = umpire::ResourceManager::getInstance();
+    alloc = rm.getAllocator("NO_OP");
+  }
+ 
+  void TearDown(benchmark::State&) {
+    alloc.getAllocationStrategy()->release();
+  }
+  
+  void NoOpAllocate(benchmark::State& st) {
+    size = static_cast<std::size_t>(st.range(0));
+    while (st.KeepRunning()) {
+      allocation = alloc.allocate(size);
+    }
+  }
+
+  void NoOpDeallocate(benchmark::State& st) {
+    while (st.KeepRunning()) {
+      alloc.deallocate(allocation);
```

Comment:
I'm not sure why this doesn't throw.  `allocation` is never assigned to something that was previously 'allocated'.  So I would assume that our allocation tracker should throw an exeption saying that you're trying to deallocate a pointer that is not currently allocated.

Reply:
IIRC, calling deallocate with an address that was not previously allocated should trigger our tracker here: https://github.com/LLNL/Umpire/blob/6d06cbce8c75bb3d970af612a770f023629f9d09/src/umpire/strategy/mixins/Inspector.cpp#L49